XLS-0040d-decentralized-identity/README.md

```diff
@@ -0,0 +1,423 @@
+
+### 4.1.1. Object Fields
+`DID` object may have the following required and optional fields. 
+
+| Field Name        |     Required?      | JSON Type | Internal Type 
+| ----------------- | :----------------: | :-------: | :-----------: 
+| `LedgerEntryType` | :heavy_check_mark: | `string`  |   `UINT16`  |
+
+Identifies the type of ledger object. The proposal recommends the value `0x0049` as the reserved entry type.
+
+---
+| Field Name        |     Required?      | JSON Type | Internal Type 
+| ----------------- | :----------------: | :-------: | :-----------: 
+| `PreviousTxnID` | :heavy_check_mark: | `string`  |   `Hash256`  |
+
+The identifying hash of the transaction that most recently modified this object.
+
+---
+| Field Name        |     Required?      | JSON Type | Internal Type 
+| ----------------- | :----------------: | :-------: | :-----------: 
+| `PreviousTxnLgrSeq` | :heavy_check_mark: | `number`  |   `UINT32`  |
+
+The index of the ledger that contains the transaction that most recently modified this object.
+
+---
+
+
+| Field Name        |     Required?      | JSON Type | Internal Type 
+| ----------------- | :----------------: | :-------: | :-----------: 
+| `URI`             |                    | `string`  |   `BLOB`    |
+
+`URI` specifies a Universal Resource Identifier, a.k.a. URI that SHOULD point to the corresponding DID document or to the data associated with the DID. This field could be an HTTP(S) URL or IPFS URI. The `URI` field is NOT checked for validity, but the field is limited to a maximum length of 256 bytes. 
+
+---
+| Field Name   |     Required?      | JSON Type | Internal Type 
+| -------------| :----------------: | :-------: | :-----------: 
+| `Data`       |                    | `string`  |   `BLOB`    |
+
+`Data` field SHOULD contain the public attestations of identity credentials associated with the DID. The `Data` field is NOT checked for validity, but the field is limited to a maximum length of 256 bytes. 
+
+---
+| Field Name   |     Required?      | JSON Type | Internal Type 
+| -------------| :----------------: | :-------: | :-----------: 
+| `DIDDocument`       |                    | `string`  |   `BLOB`    |
+
+`DIDDocument` field SHOULD contain the <a href="https://www.w3.org/TR/did-core/#did-documents"> DID document </a> per W3C standards associated with the DID. The field is NOT checked for validity, and is limited to a maximum length of 256 bytes.
+
+### 4.1.2. The `DID` Object ID Format
+`DID` object may contain public data associated with the XRPL account's identity. This requires the `DID` to be:
+- Unique in the XRPL namespace, and 
+- Uniquely associated with the XRPL account. 
+ 
+We compute the `DID` object ID, a.k.a., `DIDID`, as the `SHA-512Half` of the following values, concatenated in order:
```

Comment:
It might be less confusing to use `index` instead of `ID` here to refer to the "object ID/index"

Reply:
We've consistently used the `ID` in our specifications and documentation. To maintain consistency, I propose that we keep it as is.